src/hooks/device-orientation.js

```diff
@@ -0,0 +1,15 @@
+import { useState, useEffect } from 'react'
+
+export function useDeviceOrientation() {
+  const [state, setState] = useState(null)
+
+  useEffect(() => {
+    const handler = (event) => setState(event)
+    if (typeof window !== 'object') return
```

Comment:
I think `window` should always be defined in the useEffect calback? Were you seeing something differenet?

Reply:
Good point - maybe this would be better if it was a check for `DeviceOrientationEvent` before attaching a listener.

Reply:
it could be a good idea if it's only supported by certain browsers. I think we do that in the fullscreen hook